app/src/main/java/com/github/se/project/ui/profile/EditTutorProfile.kt

```diff
@@ -0,0 +1,221 @@
+package com.github.se.project.ui.profile
+
+import android.widget.Toast
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.runtime.snapshots.SnapshotStateList
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.PopupProperties
+import androidx.compose.ui.zIndex
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.github.se.project.model.profile.*
+import com.github.se.project.ui.components.LanguageSelector
+import com.github.se.project.ui.components.PriceSlider
+import com.github.se.project.ui.components.SubjectSelector
+import com.github.se.project.ui.navigation.NavigationActions
+
+@Composable
+fun EditTutorProfile(
+    navigationActions: NavigationActions,
+    listProfilesViewModel: ListProfilesViewModel =
+        viewModel(factory = ListProfilesViewModel.Factory)
+) {
+  val profile =
+      listProfilesViewModel.currentProfile.collectAsState().value
+          ?: return Text(
+              text = "No Profile selected. Should not happen.",
+              modifier = Modifier.testTag("editTutorNoProfile"))
+
+  val initialLanguagesList: List<Language> = profile.languages
+  val profileLanguages: SnapshotStateList<Language> = remember {
+    mutableStateListOf(*initialLanguagesList.toTypedArray())
```

Comment:
I am actually really struggling to understand that pointer reference..

Reply:
As before we had the enum i had to work around it, but now that we have switched to Listof I need to initialise the variables differently